_posts/2025-05-08-opensearch-performance-3.0.md

```diff
@@ -0,0 +1,1013 @@
+
+* **[gRPC/protobuf-based communication](https://github.com/opensearch-project/OpenSearch/issues/15190)**: As per-shard latency decreases, coordination overhead becomes more significant. Optimizations are planned to reduce serialization costs, make coordinator node communication with multiple data nodes more efficient, and support more parallelism in response handling across nodes. We're also considering adding support for a gRPC-based API using protocol buffers (protobuf) for serialization and deserialization.
+
+* **[Join support and advanced queries](https://github.com/opensearch-project/OpenSearch/issues/15185)**: Native join support is planned for 2025, with an emphasis on runtime efficiency. Techniques like index-level bloom filters or pre-join caching will be used to reduce query costs and support production-grade usage.
+
+* **[Native SIMD and vectorization](https://github.com/opensearch-project/OpenSearch/issues/9423)**: OpenSearch is exploring the use of native code and SIMD instructions beyond Lucene for operations like aggregations and result sorting. These efforts aim to reduce JVM overhead and improve throughput in CPU-bound workloads.
+
+* **[GPU acceleration for quantized vector indexes](https://github.com/opensearch-project/k-NN/issues/2295)**: OpenSearch is expanding its GPU acceleration capabilities for quantized vector indexes. While the 3.0 release introduced GPU support for full precision (fp32) indexes, the upcoming 3.1 release will extend GPU acceleration to a broader range of quantization techniques, including fp16, byte, and binary indexes. Looking beyond version 3.1, OpenSearch plans to extend GPU capabilities for search workloads.
+
+* **Disk optimized vector search---Phase 2**: OpenSearch 3.1 will introduce significant enhancements to disk-optimized vector search capabilities, marking Phase 2 of its optimization journey. Building upon the success of binary quantization, the system will incorporate the following techniques: random rotation and Asymmetric Distance Computation (ADC). These improvements, inspired by the innovative RaBitQ paper, deliver a substantial boost to recall performance while maintaining the benefits of disk optimization.
+
+* **[Partial loading of graphs using Faiss vector engine](https://github.com/opensearch-project/k-NN/issues/2401)**: Starting with version 3.1, the introduction of partial graph loading marks a major breakthrough in how vector search operates. Unlike traditional approaches that load entire graphs into memory, this new approach selectively loads only the required portions of the graph needed for search traversal. This smart loading mechanism delivers two crucial benefits: it dramatically reduces memory consumption and enables billion-scale workloads to run efficiently even on single-node environments.
```

Comment:
```suggestion
* **[Partial loading of graphs using the Faiss vector engine](https://github.com/opensearch-project/k-NN/issues/2401)**: Starting with version 3.1, the introduction of partial graph loading will mark a major breakthrough in how vector search operates. Unlike traditional approaches that load entire graphs into memory, this new approach will selectively load only the required portions of the graph needed for search traversal. This smart loading mechanism will deliver two crucial benefits: it will dramatically reduce memory consumption and enable billion-scale workloads to run efficiently even in single-node environments.
```